Modules/_sqlite/blob.c

```diff
@@ -370,6 +550,11 @@ static PyType_Slot blob_slots[] = {
     {Py_tp_clear, blob_clear},
     {Py_tp_methods, blob_methods},
     {Py_tp_members, blob_members},
+
+    // Mapping protocol
```

Comment:
It's really more like the sequence protocol. I know at the C level it's mostly the same, but would it be clearer to use the Py_sq_* slots?

Reply:
Maybe. I'll have a look.

Reply:
I think mapping protocol is more correct. We now implement the three (only) slots required by the mapping protocol. The sequence protocol was briefly mentioned in the original PR, but it was dismissed because implementing `sq_contains` could turn out to be a very slow operation. I do think that slots like `sq_concat` and `sq_inplace_concat` _might_ be useful.

We can expand the functionality by adding the sequence protocol at a later time.